### ISSUE TITLE
Crash when opening a MEC trace - Vulkan

### ISSUE THREAD

**Issue by yalcinmelihyasin** [google/gapid#2975]
GAPID Version: 1.7.0:developer
OS: linux amd64

It can be reproduced with Vulkan Bloom.

Command:
```
gapit trace -disable-pcs -disable-unknown-extensions -record-errors -no-buffer -api vulkan --start-at-frame 120 -capture-frames 30 -observe-frames 1 -out out/prefix.gfxtrace de.saschawillems.vulkanBloom
```
Stack trace: when opening prefix.gfxtrace with Gapid.
```
panic: MapMemory range ([0x00000000e8b31000-0x00000000e8b3302f]) collides with existing mapped range ([0x00000000e8b31000-0x00000000e8b3311f]) [recovered]
	panic: MapMemory range ([0x00000000e8b31000-0x00000000e8b3302f]) collides with existing mapped range ([0x00000000e8b31000-0x00000000e8b3311f])

goroutine 211 [running]:
github.com/google/gapid/core/app/crash.Crash(0x520dc40, 0xc00181bc00)
	core/app/crash/crash.go:89 +0xa7
github.com/google/gapid/core/app/crash.handler()
	core/app/crash/crash.go:56 +0x52
panic(0x520dc40, 0xc00181bc00)
	GOROOT/src/runtime/panic.go:522 +0x1b5
github.com/google/gapid/gapis/replay/builder.(*Builder).MapMemory(0xc001306f20, 0xe8b31000, 0x2030)
	gapis/replay/builder/builder.go:593 +0x42b
github.com/google/gapid/gapis/api/vulkan.externs.mapMemory(0x68061a0, 0xc0018d0930, 0x685e140, 0xc0006ee9a0, 0xf, 0xc001484150, 0xc001306f20, 0x0, 0x0, 0xc2a97a98, ...)
	gapis/api/vulkan/externs.go:65 +0x151
github.com/google/gapid/gapis/api/vulkan.(*VkMapMemory).Mutate(0xc0006ee9a0, 0x68061a0, 0xc0018d0930, 0xf, 0xc001484150, 0xc001306f20, 0x0, 0x0, 0x0, 0x0)
	bazel-out/k8-fastbuild/bin/gapis/api/vulkan/mutate.go:12614 +0xbd3
github.com/google/gapid/gapis/replay.(*adapter).MutateAndWrite(0xc0039bc340, 0x68061a0, 0xc0018d0930, 0xf, 0x685e140, 0xc0006ee9a0)
	gapis/replay/batch.go:389 +0xd7
github.com/google/gapid/gapis/api/vulkan.(*destroyResourcesAtEOS).Transform(0xc265218, 0x68061a0, 0xc0018d0930, 0xf, 0x685e140, 0xc0006ee9a0, 0x6806da0, 0xc0039bc340)
	gapis/api/vulkan/replay.go:523 +0x63
github.com/google/gapid/gapis/api/transform.TransformWriter.MutateAndWrite(...)
	gapis/api/transform/transforms.go:101
github.com/google/gapid/gapis/api/transform.(*Injector).Transform(0xc00180ce78, 0x68061a0, 0xc0018d0930, 0xf, 0x685e140, 0xc0006ee9a0, 0x68079e0, 0xc0018d0a80)
	gapis/api/transform/injector.go:39 +0x70
github.com/google/gapid/gapis/api/transform.TransformWriter.MutateAndWrite(...)
	gapis/api/transform/transforms.go:101
github.com/google/gapid/gapis/api/vulkan.(*readFramebuffer).Transform.func1(...)
	gapis/api/vulkan/read_framebuffer.go:51
github.com/google/gapid/gapis/api/vulkan.(*readFramebuffer).Transform(0xc0039bc390, 0x68061a0, 0xc0018d0930, 0xf, 0x685e140, 0xc0006ee9a0, 0x68079e0, 0xc0018d0ab0)
	gapis/api/vulkan/read_framebuffer.go:55 +0x2e4
github.com/google/gapid/gapis/api/transform.TransformWriter.MutateAndWrite(...)
	gapis/api/transform/transforms.go:101
github.com/google/gapid/gapis/api/vulkan.(*VulkanTerminator).Transform(0xc0018d09c0, 0x68061a0, 0xc0018d0930, 0xf, 0x685e140, 0xc0006ee9a0, 0x68079e0, 0xc0018d0ae0)
	gapis/api/vulkan/vulkan_terminator.go:437 +0x3dd
github.com/google/gapid/gapis/api/transform.TransformWriter.MutateAndWrite(...)
	gapis/api/transform/transforms.go:101
github.com/google/gapid/gapis/api/vulkan.(*dropInvalidDestroy).Transform(0xc0039bc380, 0x68061a0, 0xc0018d0930, 0xf, 0x685e140, 0xc0006ee9a0, 0x68079e0, 0xc0018d0b10)
	gapis/api/vulkan/replay.go:509 +0x554
github.com/google/gapid/gapis/api/transform.TransformWriter.MutateAndWrite(...)
	gapis/api/transform/transforms.go:101
github.com/google/gapid/gapis/api/vulkan.(*makeAttachementReadable).Transform(0xc265218, 0x68061a0, 0xc0018d0930, 0xf, 0x685e140, 0xc0006ee9a0, 0x68079e0, 0xc0018d0b40)
	gapis/api/vulkan/replay.go:280 +0xbf4
github.com/google/gapid/gapis/api/transform.TransformWriter.MutateAndWrite(...)
	gapis/api/transform/transforms.go:101
github.com/google/gapid/gapis/resolve/dependencygraph2.(*DCEBuilder).Flush(0xc01557c3c0, 0x68061a0, 0xc0018d0930, 0x68079e0, 0xc0018d0bd0)
	gapis/resolve/dependencygraph2/dce.go:393 +0x126
github.com/google/gapid/gapis/api/transform.Transforms.Transform(0xc01557c480, 0x7, 0xc, 0x68061a0, 0xc0018d0930, 0xc265218, 0x0, 0x0, 0x6806da0, 0xc0039bc340)
	gapis/api/transform/transforms.go:44 +0x30a
github.com/google/gapid/gapis/api/vulkan.API.Replay(0x68061a0, 0xc0018d0930, 0xc017b34a50, 0xc017b34840, 0x5950e20, 0xc017205800, 0xc01654dbc0, 0x28, 0xc017b4c600, 0x1e, ...)
	gapis/api/vulkan/replay.go:1037 +0x90d
github.com/google/gapid/gapis/replay.(*manager).execute.func1()
	gapis/replay/batch.go:330 +0x24e
github.com/google/gapid/core/app/benchmark.(*DurationCounter).Time(0xc0002b8770, 0xc017bd6118)
	core/app/benchmark/counter.go:163 +0x74
github.com/google/gapid/gapis/replay.(*manager).execute(0xc00042a080, 0x68061a0, 0xc017b34ae0, 0x6861320, 0xc000418030, 0xc529f3efbbc14a96, 0x60d6f6659aefef82, 0xcc0406ebcf532dec, 0x7b3dca2a7009b44b, 0xb18c7d30a971aca7, ...)
	gapis/replay/batch.go:326 +0xd0c
github.com/google/gapid/gapis/replay.(*manager).batch.func1(0x6861320, 0xc000418030, 0xc0006924e8, 0x7009b44bcc0406eb, 0xa971aca77b3dca2a, 0xbbc14a96b18c7d30, 0x9aefef82c529f3ef, 0xcf532dec60d6f665, 0x5950e20, 0xc017205800, ...)
	gapis/replay/batch.go:86 +0x3fa
github.com/google/gapid/gapis/replay.(*manager).batch(0xc00042a080, 0x68061a0, 0xc017b34810, 0xc014854400, 0x1e, 0x1e, 0x58ca320, 0x6555220, 0x58f8660, 0xc0173188c0, ...)
	gapis/replay/batch.go:87 +0x448
github.com/google/gapid/gapis/replay/scheduler.(*bin).exec(0xc0163e7880, 0x68061a0, 0xc0004d8210, 0xc00047a3f0)
	gapis/replay/scheduler/scheduler.go:247 +0x296
github.com/google/gapid/gapis/replay/scheduler.(*Scheduler).run(0xc00000ef40, 0x68061a0, 0xc0004b6b40)
	gapis/replay/scheduler/scheduler.go:180 +0x428
github.com/google/gapid/gapis/replay/scheduler.New.func1()
	gapis/replay/scheduler/scheduler.go:74 +0x3c
github.com/google/gapid/core/app/crash.Go.func1(0xc00000ef80)
	core/app/crash/crash.go:65 +0x43
created by github.com/google/gapid/core/app/crash.Go
	core/app/crash/crash.go:63 +0x3f

```

**Comment by AWoloszyn** [comment]
Does this happen with other traces?

**Comment by pau-baiget** [comment]
@yalcinmelihyasin please follow-up @AWoloszyn 's question

**Comment by hevrard** [comment]
I can reliably reproduce on a S9 (Exynos) with vulkan Bloom and ShadowMapping examples, I cannot on a Pixel 3.

When trying on a GLES app, I've got this other error:
```
Error: Couldn't get filter
   Cause: Context 0 is out of range [0..-1]
```

For the sake of reproduction, the commands I use are:

```
# capture
./bazel-bin/pkg/gapit trace -disable-pcs -disable-unknown-extensions -record-errors -no-buffer -api vulkan --start-at-frame 10 -capture-frames 5 -observe-frames 1 -out out/s9.shadowmapping.gfxtrace de.saschawillems.vulkanShadowmapping

# replay, this is where the error happens
./bazel-bin/pkg/gapit screenshot out/s9.shadowmapping.gfxtrace
```

**Comment by hevrard** [comment]
I can reproduce with Bloom on a Pixel2 with a regular (non-userdebug) Android build.

**Comment by hevrard** [comment]
I can reproduce without the `- observe-frames 1` flag.

**Comment by hevrard** [comment]
I cannot reproduce with an other app which is also vulkan and armeabi-v7a, so this may be a sascha willem examples specific issue.

**Comment by yalcinmelihyasin** [comment]
fixed by #3164
